lib/ClangImporter/ClangImporter.cpp

```diff
@@ -358,6 +358,39 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
       // Just use the most feature-rich C language mode.
       "-x", "c", "-std=gnu11",
     });
+
+    // The module map used for Glibc depends on the target we're compiling for,
+    // and is not included in the resource directory with the other implicit
+    // module maps. It's at {freebsd|linux}/{arch}/glibc/module.map.
+    llvm::SmallString<128> GlibcModuleMapPath;
+    if (!importerOpts.OverrideResourceDir.empty()) {
+      GlibcModuleMapPath =
+        llvm::SmallString<128>(importerOpts.OverrideResourceDir);
+    } else if (!searchPathOpts.RuntimeResourcePath.empty()) {
+      GlibcModuleMapPath =
+        llvm::SmallString<128>(searchPathOpts.RuntimeResourcePath);
+    }
+    if (!GlibcModuleMapPath.empty()) {
```

Comment:
@jrose-apple I've added the following assert:

```cpp
SmallString<128> GlibcModuleMapPath;
if (!importerOpts.OverrideResourceDir.empty()) {
  GlibcModuleMapPath = importerOpts.OverrideResourceDir;
} else if (!searchPathOpts.RuntimeResourcePath.empty()) {
  GlibcModuleMapPath = searchPathOpts.RuntimeResourcePath;
}

// Running without a resource directory is not a supported configuration.
assert(!GlibcModuleMapPath.empty());
```

The assert fired when running `test/DebugInfo/ASTSection.swift` on Linux--it turns out it was an issue with `modulewrap_main.cpp`. I've updated that code path to properly set the runtime library directory--let me know if you agree with the changes! :bow: